clients.go

```diff
@@ -5,17 +5,23 @@ import (
 	"github.com/giantswarm/microerror"
 	"github.com/giantswarm/micrologger"
 	apiextensionsclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/client-go/dynamic"
 	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
 	"k8s.io/client-go/rest"
 	"k8s.io/client-go/tools/clientcmd"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
 
 	"github.com/giantswarm/k8sclient/k8scrdclient"
 )
 
 type ClientsConfig struct {
-	Logger micrologger.Logger
+	// AddToScheme is an optional way to extend the known types to the gobal
+	// client-go scheme. Make use of it for custom CRs.
+	AddToScheme func(*runtime.Scheme) error
```

Comment:
Doesn't it make sense to make that a slice? Maybe also adding `Func` suffix to the name makes sense. Or `Funcs` if this becomes a slice.

Reply:
Let's do that when we have the use case. Or do you see that being necessary already? I would say it is kind of good that the client set is narrowed down to a certain scope. You can also just create another client when you need it for something else. I just don't know if there is a use case in our operators.

Reply:
Common use-case is reading status of one CR and reconciling another.

But now I think if we need it we can add a function like:

```go
func NewCombinedAddToScheme(addToScheme func(scheme *runtime.Scheme) error...) func(*runtime.Scheme) error {
	for _, f := range addToScheme {
		err := f(scheme)
		if err != nil {
			return microerror.Mask(err)
		}
	}
	return nil
}
```

Which would combine multiple ones. I even like that more.

Reply:
We also need this for migrations. In chart-operator we have 2 controllers. Legacy chartconfig CRD is in core. New chart CRD is in application.

For now I'll workaround by creating separate client collections.